File: backend/src/Ctor.Domain/Entities/RequiredMaterial.cs
using Ctor.Domain.Common;

namespace Ctor.Domain.Entities;

public class RequiredMaterial : BaseEntity
{
    public long BuildingId { get; set; }
    public Building Building { get; set; }
    public long MaterialId { get; set; }
    public Material Material { get; set; }
    public long Amount { get; set; }
}

File: backend/src/Ctor.ReportEngine/ReportEngine/src/main/kotlin/com/ctor/reportEngine/db/entities/RequiredMaterial.kt
package com.ctor.reportEngine.db.entities

import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.OneToOne

@Entity(name = "RequiredMaterials")
class RequiredMaterial(@Id var Id: Long, var BuildingId: Long, var MaterialId: Long, var Amount: Long) {
    @OneToOne
    @JoinColumn(name = "MaterialId", insertable = false, updatable = false)
    lateinit var Material: Material

    @ManyToOne
    @JoinColumn(name = "BuildingId", insertable = false, updatable = false)
    lateinit var Building: Building

}